File: blogeur_ia_python/blogueur_ia/serper_client.py
import httpx
from blogueur_ia.config import settings

SERPER_BASE = "https://google.serper.dev"


def _headers():
    return {"X-API-KEY": settings.serper_api_key, "Content-Type": "application/json"}


async def _post(endpoint: str, payload: dict) -> dict:
    if not settings.serper_api_key:
        return {}
    async with httpx.AsyncClient(timeout=12) as http:
        try:
            resp = await http.post(f"{SERPER_BASE}/{endpoint}", headers=_headers(), json=payload)
            if resp.status_code != 200:
                return {}
            return resp.json()
        except Exception:
            return {}


async def serper_search(q: str, num: int = 8) -> list[dict]:
    data = await _post("search", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl})
    return data.get("organic", [])


async def serper_search_time_define(q: str, num: int = 8, tbs: str = "qdr:w") -> list[dict]:
    data = await _post("search", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl, "tbs": tbs})
    return data.get("organic", [])


async def serper_news(q: str, num: int = 6) -> list[dict]:
    data = await _post("news", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl})
    return data.get("news", [])


async def serper_images(q: str, num: int = 6) -> list[dict]:
    data = await _post("images", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl})
    return data.get("images", [])


async def serper_videos(q: str, num: int = 5) -> list[dict]:
    data = await _post("videos", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl})
    return data.get("videos", [])


async def serper_places(q: str, num: int = 5) -> list[dict]:
    data = await _post("places", {"q": q, "num": num, "gl": settings.serper_gl, "hl": settings.serper_hl})
    return data.get("places", [])

File: blogeur_ia_node/src/serperClient.ts
import axios from "axios";
import { settings } from "./config";

const SERPER_BASE = "https://google.serper.dev";

async function post(endpoint: string, payload: Record<string, any>): Promise<any> {
  if (!settings.serperApiKey) return {};
  try {
    const resp = await axios.post(`${SERPER_BASE}/${endpoint}`, payload, {
      headers: { "X-API-KEY": settings.serperApiKey, "Content-Type": "application/json" },
      timeout: 12000,
    });
    return resp.status === 200 ? resp.data : {};
  } catch {
    return {};
  }
}

const baseParams = () => ({ gl: settings.serperGl, hl: settings.serperHl });

export async function serperSearch(q: string, num = 8): Promise<any[]> {
  const data = await post("search", { q, num, ...baseParams() });
  return data.organic || [];
}

export async function serperSearchTimeDefine(q: string, num = 8, tbs = "qdr:w"): Promise<any[]> {
  const data = await post("search", { q, num, tbs, ...baseParams() });
  return data.organic || [];
}

export async function serperNews(q: string, num = 6): Promise<any[]> {
  const data = await post("news", { q, num, ...baseParams() });
  return data.news || [];
}

export async function serperImages(q: string, num = 6): Promise<any[]> {
  const data = await post("images", { q, num, ...baseParams() });
  return data.images || [];
}

export async function serperVideos(q: string, num = 5): Promise<any[]> {
  const data = await post("videos", { q, num, ...baseParams() });
  return data.videos || [];
}

export async function serperPlaces(q: string, num = 5): Promise<any[]> {
  const data = await post("places", { q, num, ...baseParams() });
  return data.places || [];
}